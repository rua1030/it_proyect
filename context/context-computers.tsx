'use client';

import React, {
  createContext, useState, useContext, ReactNode,
} from 'react';

const initialSupplies = [
  {
    id: 1, brand: 'Dell', serial: 'DELL123456', status: 'Available',
  },
  {
    id: 2, brand: 'HP', serial: 'HP789012', status: 'In use',
  },
  {
    id: 3, brand: 'Lenovo', serial: 'LEN345678', status: 'In storage',
  },
  {
    id: 4, brand: 'Apple', serial: 'APP901234', status: 'Available',
  },
  {
    id: 5, brand: 'Asus', serial: 'ASUS567890', status: 'Stolen',
  },
];

const supplyStatuses = ['Available', 'In storage', 'In use', 'Stolen'];

interface SupplyContextProps {
  status: typeof supplyStatuses;
  supplies: typeof initialSupplies;
  filter: string;
  currentPage: number;
  suppliesPerPage: number;
  addSupply: () => void;
  editSupply: (id: number) => void;
  deleteSupply: () => void;
  confirmDelete: (supply: any) => void;
  confirmadd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToDelete: any;
  setSupplyToDelete: React.Dispatch<React.SetStateAction<any>>;
  addModal: boolean;
  setaddModal: React.Dispatch<React.SetStateAction<boolean>>;
  supplyToadd: any;
  setSupplyToadd: React.Dispatch<React.SetStateAction<any>>;
}

const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderComputer({
  children,
}: {
  children: ReactNode;
}) {
  const [supplies, setSupplies] = useState(initialSupplies);
  const [filter, setFilter] = useState('');
  const [currentPage, setCurrentPage] = useState(1);
  const [suppliesPerPage] = useState(5);
  const [deleteModal, setDeleteModal] = useState(false);
  const [addModal, setaddModal] = useState(false);
  const [supplyToDelete, setSupplyToDelete] = useState(null);
  const [supplyToadd, setSupplyToadd] = useState(null);
  const [status] = useState(supplyStatuses);

  // Simulate adding a supply
  const addSupply = () => {
    console.log('Add supply');
  };

  // Simulate editing a supply
  const editSupply = (id: number) => {
    console.log('Edit supply', id);
  };

  // Open confirmation modal for deletion
  const confirmDelete = (supply: any) => {
    setSupplyToDelete(supply);
    setDeleteModal(true);
  };

  const confirmadd = () => {
    setaddModal(true);
  };

  // Delete supply
  const deleteSupply = () => {
    if (supplyToDelete) {
      setSupplies(supplies.filter((supply) => supply.id !== supplyToDelete.id));
      setDeleteModal(false);
      setSupplyToDelete(null);
    }
  };

  // Simulate status change
  const changeStatus = (id: number, newStatus: string) => {
    setSupplies(
      supplies.map((supply) => (supply.id === id ? { ...supply, status: newStatus } : supply)),
    );
  };

  return (
    <SupplyContext.Provider
      value={{
        status,
        supplies,
        filter,
        currentPage,
        suppliesPerPage,
        addSupply,
        editSupply,
        deleteSupply,
        confirmDelete,
        confirmadd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        supplyToDelete,
        setSupplyToDelete,
        addModal,
        setaddModal,
        supplyToadd,
        setSupplyToadd,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}

export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
