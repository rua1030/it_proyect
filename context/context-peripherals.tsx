'use client';

import React, {
  createContext, useState, useContext, ReactNode,
} from 'react';

const initialPeripherals = [
  {
    id: 1,
    brand: 'Logitech',
    name: 'MX Master 3',
    quantity: 50,
    status: 'In stock',
  },
  {
    id: 2,
    brand: 'Dell',
    name: 'P2419H Monitor',
    quantity: 30,
    status: 'Low stock',
  },
  {
    id: 3,
    brand: 'Corsair',
    name: 'K70 RGB Keyboard',
    quantity: 20,
    status: 'Out of stock',
  },
  {
    id: 4,
    brand: 'Jabra',
    name: 'Evolve 75 Headset',
    quantity: 40,
    status: 'In stock',
  },
  {
    id: 5,
    brand: 'Microsoft',
    name: 'Surface Dock',
    quantity: 15,
    status: 'Low stock',
  },
];

const peripheralStatuses = ['In stock', 'Low stock', 'Out of stock'];

interface SupplyContextProps {
  statuses: typeof peripheralStatuses;
  peripherals: typeof initialPeripherals;
  filter: string;
  currentPage: number;
  addPeripheral: () => void;
  editPeripheral: (id: number) => void;
  deletePeripheral: () => void;
  confirmDelete: (supply: string) => void;
  confirmadd: () => void;
  changeStatus: (id: number, newStatus: string) => void;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  peripheralToDelete: string;
  setPeripheralToDelete: React.Dispatch<React.SetStateAction<string>>;
  addModal: boolean;
  setaddModal: React.Dispatch<React.SetStateAction<boolean>>;
  peripheralToadd: string;
  setPeripheralToadd: React.Dispatch<React.SetStateAction<string>>;
}

const SupplyContext = createContext<SupplyContextProps | undefined>(undefined);

export function SupplyProviderPeripherals({
  children,
}: {
  children: ReactNode;
}) {
  const [peripherals, setPeripherals] = useState(initialPeripherals);
  const [filter, setFilter] = useState('');
  const [currentPage] = useState(1);
  const [deleteModal, setDeleteModal] = useState(false);
  const [peripheralToDelete, setPeripheralToDelete] = useState(null);
  const [addModal, setaddModal] = useState(false);
  const [peripheralToadd, setPeripheralToadd] = useState(null);
  const [statuses] = useState(peripheralStatuses);

  const addPeripheral = () => {
    console.log('Add peripheral');
  };

  // Simulate editing a peripheral
  const editPeripheral = (id: number) => {
    console.log('Edit peripheral', id);
  };

  // Open confirmation modal for deletion
  const confirmDelete = () => {
    setDeleteModal(true);
  };

  const confirmadd = () => {
    setaddModal(true);
  };

  // Delete peripheral
  const deletePeripheral = () => {
    if (peripheralToDelete) {
      setPeripherals(
        peripherals.filter(
          (peripheral) => peripheral.id !== peripheralToDelete.id,
        ),
      );
      setDeleteModal(false);
      setPeripheralToDelete(null);
    }
  };

  // Simulate status change
  const changeStatus = (id: number, newStatus: string) => {
    setPeripherals(
      peripherals.map((peripheral) => (peripheral.id === id ? { ...peripheral, status: newStatus } : peripheral)),
    );
  };
  return (
    <SupplyContext.Provider
      value={{
        peripherals,
        filter,
        currentPage,
        addPeripheral,
        editPeripheral,
        deletePeripheral,
        confirmDelete,
        confirmadd,
        changeStatus,
        setFilter,
        deleteModal,
        setDeleteModal,
        peripheralToDelete,
        setPeripheralToDelete,
        addModal,
        setaddModal,
        peripheralToadd,
        setPeripheralToadd,
        statuses,
      }}
    >
      {children}
    </SupplyContext.Provider>
  );
}
export const useSupplyContext = () => {
  const context = useContext(SupplyContext);
  if (!context) {
    throw new Error('useSupplyContext must be used within a SupplyProvider');
  }
  return context;
};
