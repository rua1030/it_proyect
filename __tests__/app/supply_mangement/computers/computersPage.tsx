import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { SupplyProviderComputer } from '@/context/context-computers';
import SupplyManagement from '@/app/supply_management/computers/page';

describe('ComputersManagement', () => {
  const setup = () => {
    render(
      <SupplyProviderComputer>
        <SupplyManagement />
      </SupplyProviderComputer>,
    );
  };

  it('matches snapshot', () => {
    const { asFragment } = render(
      <SupplyProviderComputer>
        <SupplyManagement />
      </SupplyProviderComputer>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all necessary fields', () => {
    setup();
    expect(
      screen.getByText('Supply Management / Laptops'),
    ).toBeInTheDocument();
    expect(screen.getByText('Brand')).toBeInTheDocument();
    expect(screen.getByText('Serial')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Actions')).toBeInTheDocument();
    expect(screen.getByText('Previous')).toBeInTheDocument();
    expect(screen.getByText('Next')).toBeInTheDocument();
  });

  it('renders Computers Management title', () => {
    setup();
    expect(
      screen.getByText('Supply Management / Laptops'),
    ).toBeInTheDocument();
  });

  it('updates filter on input change', () => {
    setup();
    const input = screen.getByPlaceholderText('Search by brand or serial...');
    fireEvent.change(input, { target: { value: 'Dell' } });
    expect(input).toHaveValue('Dell');
  });

  it('opens add computer modal on button click', () => {
    setup();
    const addButton = screen.getByText('Add Laptops');
    fireEvent.click(addButton);
    expect(screen.getByText('add computer')).toBeInTheDocument();
  });

  it('changes page when pagination button is clicked', () => {
    setup();
    const nextPageButton = screen.getByText(/Next/i);
    fireEvent.click(nextPageButton);
  });
});
