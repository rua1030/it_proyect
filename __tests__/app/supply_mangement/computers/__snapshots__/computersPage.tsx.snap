// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComputersManagement matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto p-4"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      Supply Management / Laptops
    </h1>
    <div
      class="flex justify-between items-center mb-4"
    >
      <input
        class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 max-w-sm"
        placeholder="Search by brand or serial..."
        type="text"
        value=""
      />
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      >
        <svg
          class="lucide lucide-plus mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
         Add Laptops
      </button>
    </div>
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Brand
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Serial
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Status
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              Dell
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              DELL123456
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r0:"
                aria-expanded="false"
                class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Available
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-square-pen h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                  />
                </svg>
              </button>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-trash2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              HP
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              HP789012
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r1:"
                aria-expanded="false"
                class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  In use
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-square-pen h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                  />
                </svg>
              </button>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-trash2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              Lenovo
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              LEN345678
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r2:"
                aria-expanded="false"
                class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  In storage
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-square-pen h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                  />
                </svg>
              </button>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-trash2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              Apple
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              APP901234
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r3:"
                aria-expanded="false"
                class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Available
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-square-pen h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                  />
                </svg>
              </button>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-trash2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              Asus
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              ASUS567890
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r4:"
                aria-expanded="false"
                class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-[180px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Stolen
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr 1"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-square-pen h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                  />
                  <path
                    d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                  />
                </svg>
              </button>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              >
                <svg
                  class="lucide lucide-trash2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="flex justify-center mt-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
        disabled=""
      >
        Previous
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mx-1"
      >
        1
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 ml-2"
        disabled=""
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
